### PR TITLE
Added support for extra mouse buttons

### LINKED ISSUE #3031: [Feature Request] Mouse extra buttons
WPF supports the extra mouse buttons (`XButton1` and `XButton2`) with the `MouseDown` event. I believe Avalonia should have an API to handle these buttons. I think this would be part of the `PointerPoint`/`PointerPointProperties` API.

For reference: [WPF MouseButton enum](https://docs.microsoft.com/en-us/dotnet/api/system.windows.input.mousebutton)

### PR DESCRIPTION
**Pull request by jp2masa**
## Changes
- Added support for extra mouse buttons.
- Fixed a typo in `X11Window` (`Button2Mask` -> `Button3Mask`).

## Fixed issues
Fixes #3031

Only tested on Windows.

### CODE REVIEW

File: src/Avalonia.Input/PointerPoint.cs [hunk 1]
```diff
@@ -67,9 +81,13 @@ public enum PointerUpdateKind
         LeftButtonPressed,
         MiddleButtonPressed,
         RightButtonPressed,
+        XButton1Pressed,
```
Comment (kekekeks) [inline src/Avalonia.Input/PointerPoint.cs]: That breaks binary compatibility. Not sure if we care at the current point

File: src/Avalonia.Input/Raw/RawPointerEventArgs.cs [hunk 2]
```diff
@@ -14,6 +14,10 @@ public enum RawPointerEventType
         RightButtonUp,
         MiddleButtonDown,
         MiddleButtonUp,
+        XButton1Down,
```
Comment (kekekeks) [inline src/Avalonia.Input/Raw/RawPointerEventArgs.cs]: Same here, values starting from `Move` will have old values in dependent assemblies compiled before this change

File: src/Avalonia.X11/X11Window.cs [hunk 3]
```diff
@@ -342,10 +342,17 @@ void OnEventSync(XEvent ev)
             {
                 if (ActivateTransientChildIfNeeded())
                     return;
-                if (ev.ButtonEvent.button < 4)
-                    MouseEvent(ev.ButtonEvent.button == 1 ? RawPointerEventType.LeftButtonDown
-                        : ev.ButtonEvent.button == 2 ? RawPointerEventType.MiddleButtonDown
-                        : RawPointerEventType.RightButtonDown, ref ev, ev.ButtonEvent.state);
+                if (ev.ButtonEvent.button < 4 || ev.ButtonEvent.button == 8 || ev.ButtonEvent.button == 9)
```
Comment (kekekeks) [inline src/Avalonia.X11/X11Window.cs]: On systems with modern X server mouse input is handled by [XI2Manager](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.X11/XI2Manager.cs) class.

File: src/Avalonia.Input/PointerPoint.cs (hunk 1 above)
Comment (grokys) [reply src/Avalonia.Input/PointerPoint.cs]: Yeah I don't think we care at this point. 0.9 is going to have a number of breaking changes so everything is going to need to be recompiled anyway.

**Review by kekekeks — APPROVED** [review]

**Comment by grokys** [comment]
@jp2masa the code is currently not compiling according to CI.

**Comment by jp2masa** [comment]
I know, I used C# 8 switch expression, which is much easier to read (current code uses ternary inside ternary). This would require .NET Core 3.0 SDK.

**Pull request merged**